docs/design/compiler-generated-code-handling.md

```diff
@@ -0,0 +1,336 @@
+# Handling of compiler generated code
+
+Modern compilers provide language features which require lot of fancy code generation by the compiler. Not just pure IL generation, but producing new types, methods and fields. An example is `async`/`await` in C# which turns the body of the method into a separate class which implements a state machine.
+
+Lot of the trimming logic relies on attributes authored by the developer. These provide hints to the trimmer especially around ares which are otherwise problematic, like reflection. For example see [reflection-flow](reflection-flow.md) for an example of such attribute.
+
+## Problem
+
+User authored attributes are not propagated to the compiler generated code. For example:
```

Comment:
Which languages suffer from this problem?

Reply:
C# - is the one I know of. VB probably as well (didn't try).
While I agree that we should try to come up with solutions which could work for all languages, I think it's OK to start by focusing on C# first.

Reply:
I don't have a problem focusing on C# but the wording used in the design doc should indicate that more clearly.